doc/lint-lifecycle.md

```diff
@@ -0,0 +1,66 @@
+# The Life of a Lint
+
+## Lint States
+
+### Proposed
+
+A lint begins its life as a **proposal** which is _pending_ until _accepted_.
+
+### Accepted
+
+After discussion and sufficient agreement within the team, a proposed lint becomes **accepted**.
+
+An accepted lint is ready for implementation.
+
+### Experimental
+
+Lints can be marked experimental. 
+
+Reasons a lint might be marked experimental include a lint
+
+* being tentatively introduced
+* having unknown value
+* having incomplete semantics
+* having outstanding (fixable) false positives or
+* that might be known to be temporary.
+
+### Stable
+
+Stable lints are lints whose semantics and implementation are considered complete. False
+positives (that are not known, accepted and documentted limitations of lint semantics) are
+considered bugs and fixing them should be prioritized. False negatives may be bugs or
+enhancements (depending on semantics).
+
+### Deprecated
+
+Deprecated lints are lints that we plan to remove.
+
+Reasons for deprecation include:
+
+* semantics that don't make sense with current language semantics (e.g., null-safety)
+* stale advice
+* poor performance or
+* poor user experience (e.g., too many false positives).
+
+### Removed
+
+Lints we no longer support are removed.
+
+In general removal is preceded by a period of deprecation.
+
+## State Transitions
+
+### Deprecation
```

Comment:
Should this be moved under "Deprecated"? It feels like a discussion about how lints get moved to that state, so it might be something we can combine.

Also, do we want to move lints from experimental to deprecated before removing them, or is the ability to remove them without notice part of what makes them experimental?